ignite/metrics/ssim.py

```diff
@@ -13,7 +13,10 @@ class SSIM(Metric):
     """
     Computes Structural Similarity Index Measure
 
-    - ``update`` must receive output of the form ``(y_pred, y)``.
+    - ``update`` must receive output of the form ``(y_pred, y)``. They have to be of the same type.
+        Valid :class:`torch.dtype` are the following:
+        - on CPU: `torch.float32`, `torch.float64`.
```

Comment:
Can you please check if torch.bfloat16 is supported  ? 
https://pytorch.org/docs/stable/tensors.html

Reply:
I already tried on GPU it didn't work, will try on CPU :)

Reply:
On CPU:
- RuntimeError: "reflection_pad2d" not implemented for 'Half'
- RuntimeError: "reflection_pad2d" not implemented for 'BFloat16

On GPU:
- numpy doesn't support Bfloat16 (yet), so there is no baseline comparison. Below are the distribution of the difference between float16 or bloat16 in torch and float16 in scikit-image. So it is possible to create a special case for bfloat16 where it is tested against a numpy float16 array.

![image](https://github.com/pytorch/ignite/assets/50196352/aba7b6dd-6b9e-4726-aeff-537fe084580b)
![image](https://github.com/pytorch/ignite/assets/50196352/9633bd90-8d50-4b9f-8b50-9c3fdaf21e6d)

Reply:
Thanks for the analysis, yes, it makes sense to raise atol value to ~1e-3 for float16/bfloat16 dtypes due to their low-precision.

Reply:
Here is the scale comparison against all types on CPU and GPU.

![image](https://github.com/pytorch/ignite/assets/50196352/919a7168-8805-410c-8b9d-c80c70264025)

I will add a parameter in my test function for the precision that will be linked to the dtype.

Reply:
Would it make more sense to plot absolute difference instead of "difference value" ?